lib/dialects/abstract/query-generator.js

```diff
@@ -1456,7 +1456,7 @@ const QueryGenerator = {
         // The main model attributes is not aliased to a prefix
         joinOn = `${this.quoteTable(parent.as || parent.model.name)}.${this.quoteIdentifier(attrLeft)}`;
       } else {
-        joinOn = this.quoteIdentifier(`${asLeft}.${attrLeft}`);
+        joinOn = this.quoteIdentifier(`${asLeft.replace('->', '.')}.${attrLeft}`);
```

Comment:
Not in the tests I've done, but effectively I think it could contains more than only one ->